zerver/lib/actions.py

```diff
@@ -5485,6 +5485,38 @@ class DeleteMessagesEvent(TypedDict, total=False):
     stream_id: int
 
 
+class UpdateMessagesEvent(TypedDict, total=False):
+    type: str
+    user_id: int
+    edit_timestamp: int
+    message_id: int
+    message_ids: List[int]
+
+    propagate_mode: Optional[str]
+    orig_content: Optional[str]
+    orig_rendered_content: Optional[str]
+    prev_rendered_content_version: Optional[int]
+    content: Optional[str]
+    flags: Optional[List[str]]
+    is_me_message: Optional[bool]
+    rendered_content: Optional[str]
+
+    push_notify_user_ids: Optional[List[int]]
+    stream_push_user_ids: Optional[List[int]]
+    stream_email_user_ids: Optional[List[int]]
+    prior_mention_user_ids: Optional[List[int]]
+    mention_user_ids: Optional[List[int]]
+    presence_idle_user_ids: Optional[List[int]]
+    wildcard_mention_user_ids: Optional[List[int]]
+
+    stream_name: Optional[str]
+    stream_id: Optional[int]
+    new_stream_id: Optional[int]
+    orig_subject: Optional[str]
+    subject: Optional[str]
+    topic_links: Optional[List[Dict[str, str]]]
```

Comment:
This is kinda giant; I wonder if there's a good way to define simpler classes; e.g. have the common elements in a base class, and then subclasses for "has topic edit"?  Not sure; in any case, it's an improvement to just have a defined type even if it is giant.

Reply:
Also, ideally we'd use the class in `zerver/tornado_event_queue.py` if possible.

Reply:
> This is kinda giant; I wonder if there's a good way to define simpler classes; e.g. have the common elements in a base class, and then subclasses for "has topic edit"? Not sure; in any case, it's an improvement to just have a defined type even if it is giant.
I initially tried to do it this way -- but there are total 4 branching `ifs` - if there is content, if it is a stream message, if the topic is changed or stream is moved, if the topic is changed - giving rise to 16 possible classes. That *might* make the class structure a little cluttered in my opinion.
Also, in TypedDict, it is not possible to change type of an inherited parameter in the child class, so an approach where we have Optional elements made non-optional in child classes won't work out as well.

Reply:
@timabbott Would this large number of different classes be desirable? Since this is the only way I think we could remove the `total=False` if that is desired.